### PR TITLE
Team home: parent dashboard with the next three events, one-tap RSVP, and kid chart summary

### LINKED ISSUE #48: Team home: parent dashboard with next event, one-tap RSVP, and kid chart summary
## Problem

After accepting an invite, a parent lands on `/t/[teamId]` and sees the season label, the allPlay setting, and coach contact cards — nothing else (`src/app/t/[teamId]/page.tsx`). The three questions the brief says the app exists to answer (when, where, is my kid playing) all live two or more taps deeper, and RSVPing currently takes 4 taps + 5 full page loads via Schedule → event page.

## Proposal

Make team home answer everything at a glance for a parent:

1. **Next event card** — the soonest upcoming event (game *or* practice; unlike readiness this is informational, so practices belong here): type, date/time via `formatEventDateTime`, location, notes, linking to `/schedule/[eventId]`.
2. **One-tap RSVP** — for each kid the viewer guards on this team (`guardedRosteredPlayerIds` in `src/lib/rsvps.ts`), render the same Going / Not going forms the event page uses, posting to the existing `rsvpAction`. Redirect back to team home so the parent never leaves the page.
3. **&#34;Your kid&#34; summary line per guarded kid** — e.g. `Reese · #12 · Bats 3rd · SS`, from `RosterEntry.battingOrder` / `position` + `POSITION_LABELS`; on an allPlay team a null position reads `OF`, on a selective team `Bench`. Reuse the `chartRole()` shape from `readiness/page.tsx` if it can be lifted into a shared pure helper.

Coaches/owner keep their current view (contact card already hides for them); the dashboard sections simply render empty-quietly for members guarding no players.

All composition — the queries already exist (`nextGame` in `src/lib/schedule.ts`, `guardedRosteredPlayerIds`, chart columns). No schema changes. Respect the pattern rules: Prisma only via `src/lib/`, pure decision logic tested without a database.

## Acceptance criteria

- [x] A parent opening team home sees the next event (game or practice) with time, place, and a link to the event page
- [x] They can RSVP each of their kids in one tap from this page, with the current state visible
- [x] Each guarded kid shows jersey, batting slot, and position (or Bench/OF) in one line
- [x] Archived teams show the summary but no RSVP buttons (write-rejection copy problem must not reappear here)
- [x] Coaches&#39; view is unchanged apart from any next-event card we choose to show them too

_From the Aug 2026 UX audit (finding P1 in the Dugout Report)._

---

**Delivered by #63.** Scope was extended during implementation: team home shows the **next three** events, each independently answerable, rather than a single next-event card. See the closing comment for per-criterion verification.

### PR DESCRIPTION
**Pull request by BRCornelius**
## Summary

Team home showed a parent the season label and the coach's phone number and nothing else, so the three questions the brief says this app exists to answer — when, where, is my kid playing — were all two or more taps away, and RSVPing took four taps and five page loads. This PR makes `/t/[teamId]` answer all three at a glance: each kid's chart line, then the next three events, every one of them answerable in one tap without leaving the page. Closes #48.

No schema changes. Planning docs are archived to `.agents/archive/issue-planner/48/`; the original proposal is [on the issue](https://github.com/Corneliuses/baseball/issues/48#issuecomment-5374118733) — note it describes a single next-event card, which a later commit supersedes.

**This PR also gives the repo its first CI pipeline** — see *Beyond the issue* below.

## Key Decisions

**New `nextEvents`, rather than loosening `nextGame`.** The cards need practices; `nextGame` is games-only by deliberate contract, because readiness and `/view` check a chart a practice doesn't have. A flag on one shared function would let a caller ask for the wrong rule by passing the wrong boolean, so `selectNextEvents` is the informational twin, `selectNextEvent` is its `limit: 1` case, and `selectNextGame` delegates its grace window to it — "has this finished yet" and "which is soonest" are each defined once, so a page showing three events and a page showing one cannot disagree about which is next.

**The buttons sit under the event, not under the child.** Grouping them the other way made every event print its name and date once per kid, and the event is the unit of the question anyway — "who's coming Saturday" is what the event page already asks. The chart line moved above the cards, where one line per kid means a parent with one child gets all three answers in a screenful.

**One RSVP read for all three events, bucketed by a pure function.** Asking the same question three times over separately fetched rows is how two cards end up disagreeing about the same family. Rows carry their `eventId`; `buildRsvpStateMapsByEvent` gives every requested event a map and every player an entry, so a lookup never misses.

**`from` is an enum, never a URL.** Both surfaces that RSVP post to one action, and the only thing they disagree about is where the parent was standing. A `returnTo` URL field would say the same thing while handing anyone who can craft a form an open redirect out of a signed-in POST. `z.enum(["home"])` cannot express a destination `actions.ts` did not already write down; anything unrecognised falls back to the event page.

**The start-time gate is enforced twice, and it disambiguates rather than authorizes.** Team home hides its buttons once an event starts; `rsvpAction` re-checks for `from=home` posts, because a dashboard left open through first pitch still holds a form that submits. It is deliberately origin-scoped: the event page has always recorded a late answer on purpose — a parent realising at 9:15 they can't make the 9:00 game is telling the coach something useful — and what makes it wrong from home is that the event there is *page-selected*, so on a doubleheader morning "Not going" may well mean the noon game.

**`chartRole` lifted rather than copied**, into `src/lib/chart-role.ts`, with the bench label opt-in: readiness keeps printing nothing for an unplaced player, because a coach reading a list of who is out already knows the chart. Two copies would drift the way the two diamonds' names did before `buildDiamondNames`.

**Sections key on guardianship, not role.** A coach who is also a parent answers for their own kid here, exactly as they already can from the event page. A member guarding nobody gets no summary and no buttons, and neither query runs.

## What's in Scope

- The next three events, each as a card with time, mapped location, notes, a details link, and its own RSVP rows; a quiet empty state when nothing is scheduled
- One-tap Going / Not going per kid **per event**, with the current answer visible, staying on team home; each button's `aria-label` names the child *and* the event's date and time, since two practices would otherwise give every control the same accessible name
- A one-line `Reese · #12 · Bats 3rd · SS` summary per guarded kid, with `OF` on an allPlay team and `Bench` on a selective one
- Archived teams keep the summaries and any answer already given, and lose the buttons
- `hasChartSet` consolidated into `chart-view.ts`, replacing the copy there and the one on the readiness page

### Out of Scope

- Per-game lineup data of any kind — Decision 16 stands, and readiness stays coach-only at `/readiness`
- Notifications about upcoming events
- Schedule-list and event-page UI, beyond the `from` parameter on the shared action
- Making the event count configurable — `UPCOMING_LIMIT = 3` is deliberately a constant
- Hardening the older `ERROR_MESSAGES` tables under `schedule/` and `roster/` against the prototype lookup below; same shape, pre-existing, recorded in `AGENTS.md` instead of widening this PR

## Beyond the issue: a CI pipeline

Added at the maintainer's request while finalizing, because the gap surfaced here: **the only check on this PR was Vercel's preview-comment bot**, so "green" meant a deploy comment had posted. Nothing verified that a branch lints, typechecks, tests, or compiles. Every PR carried that blind spot.

`.github/workflows/ci.yml` adds two parallel jobs on every PR and every push to `main`: `check` (lint, typecheck, unit tests as separately named steps, each `if: !cancelled()` so one push reports every problem) and `build`.

**The build step runs `pnpm exec next build`, never `pnpm build`.** The package script is `prisma migrate deploy && next build` — right on Vercel, dangerous in CI, because Preview and Production share one `DATABASE_URL`: a CI job running the migrate step would apply migrations to the **production database** on every pull request, from any branch, including one whose migrations are still being written. Verified nothing is lost by skipping it — `next build` completes with no `DATABASE_URL` and no `.env` at all, because every route under `/t/[teamId]` is server-rendered on demand. `package.json` gains `packageManager`, which `pnpm/action-setup` requires.

## Bugs Found and Fixed During Self-Review

Re-reading the diff caught two before review: `getChart` is a `findMany` with no `orderBy`, so siblings would swap places between requests — `byJerseyThenName` is now exported rather than hand-rolled a third time — and an unreachable `chartLine === ""` branch. The first ordering test passed trivially because its fixture was already sorted; it now hands the rows over reversed and fails without the sort.

A `/code-review` pass surfaced four more, then the Copilot review surfaced five. Each fix has a regression test **verified to fail against the pre-fix code**:

| What was wrong | Why it was wrong | Fix |
|---|---|---|
| `Bench` on any selective-team kid with no position, batting order included | A kid batting third with no fielding spot is *in the order*. `/view`'s bench filter documents exactly this, so home said "Bats 3rd · Bench" while `/view` showed that kid third and on no bench | `benchLabel` applies under the same condition. The test covering this had pinned the wrong string |
| No `hasChart` check | A team with no chart got a definite spot for every kid — `OF` or `Bench` — where `/view` says "No chart set yet" | `hasChartSet` gates the line, replacing two existing copies of the predicate |
| Start-time gate lived only in the render | `requireGuardedEvent` never checked `startsAt`, so a dashboard left open through first pitch could still post | `rsvpAction` re-checks for `from=home`, redirecting to `?error=event-started` |
| `ERROR_MESSAGES` was a plain object literal | `?error=constructor` resolves an `Object.prototype` member — truthy, so `??` never fires — and React throws `Functions are not valid as a React child` | `Object.assign(Object.create(null), {…})`, matching `/profile` |
| aria-labels named only the kind of event | Two practices gave four buttons two accessible names between them | Labels carry the event's date and time |
| `listRsvpsForEvents` had no wrapper tests | Its sibling pins team scope and columns; losing `eventId` from the select would collapse every card onto one answer | Four tests: filter, team scope, columns, empty-input short-circuit, error propagation |

Verifying the acceptance criteria during finalization caught one more: a test named *"shows the card to a coach as well as a parent"* overrode the role to `PARENT` and asserted only that — it never checked the half it was named for. It now renders under both roles.

## Known Gaps / Follow-ups

- Nothing here has been exercised against real Postgres or on a real phone.
- A parent with several kids and three events sees a fairly long page. Nothing about it is wrong, but it's worth looking at on a real phone before the season starts.
- The actions suite was time-dependent (a past-dated fixture) until this PR pinned its clock. Other suites may carry the same latent issue; a future time-sensitive feature would trip it.

## Test Plan

- [ ] CI: `check` and `build` jobs green on this PR — the first PR in this repo where that means anything.
- [ ] `pnpm check` locally — **82 files, 1181 tests passing**.
- [ ] `pnpm dev`, sign in as a parent with a kid on a team that has a chart and three upcoming events: the kid's line reads `Reese · #12 · Bats Nth · <position>`, three event cards follow with time, place and a working details link, and tapping **Going** on the third card returns to `/t/<teamId>?saved=1` with only that card's button filled.
- [ ] Add a practice sooner than the next game: it takes the first card. Confirm `/readiness` and `/view` still name the *game*.
- [ ] Edge cases covered by tests: an event in progress loses its buttons while a later one keeps them; three events hold three independent answers; two kids get separate buttons on every card; a kid batting third with no position never reads "Bench"; a team with no chart reads "No chart set yet"; an archived team keeps summaries and loses buttons; a member guarding nobody triggers no queries; siblings sort by jersey; `?error=constructor` falls back instead of crashing.

### CODE REVIEW

File: src/app/t/[teamId]/schedule/actions.ts [hunk 1]
```diff
@@ -255,29 +300,38 @@ async function requireGuardedEvent(teamId: string, eventId: string, playerId: st
...
   if (!parsedResponse.success) {
-    redirect(`/t/${teamId}/schedule/${eventId}?error=invalid-rsvp`);
+    redirect(rsvpReturnUrl(origin, teamId, eventId, "?error=invalid-rsvp"));
   }
 
   try {
     // Write against the id `getEvent` resolved, not the raw form field — same
     // convention as the roster actions deriving playerId from getRosterEntry.
-    const event = await requireGuardedEvent(teamId, eventId, playerId);
+    const event = await requireGuardedEvent(teamId, eventId, playerId, origin);
```
[the remark this answers is not in the document]
Comment (BRCornelius) [reply src/app/t/[teamId]/schedule/actions.ts:339]: Right conclusion, fixed in 7871cb8 — but worth being precise about what it was.

You're correct that `requireGuardedEvent` proved team access, event-on-team and guardianship and never looked at `startsAt`, so the render was the only thing enforcing the gate. That's the mistake `AGENTS.md` already warns about for Proxy ("only the action knows what is being written"), and the realistic case isn't a crafted POST — it's a parent who left the dashboard open through first pitch and taps the form it already rendered. `rsvpAction` now re-checks and redirects to `?error=event-started`.

I did scope it to `from=home` as you suggested, though for a different reason than "origin-aware" implies — this **disambiguates rather than authorizes**. The event page has always recorded a late answer deliberately: a parent realising at 9:15 they can't make the 9:00 game is telling the coach something useful, and readiness still surfaces that game via `GAME_GRACE_MS`. What makes it wrong from home is that the event there is *page-selected*, so on a doubleheader morning "Not going" may well mean the noon game. That also means refusing an origin-less POST would protect nothing, since that's exactly what the event page sends — so there's no bypass here, just a rule that applies to the surface that needs it.

Three tests, verified to fail without the guard: a started event from home is refused, an event starting in one second is still accepted, and the event page still records a late answer. The actions suite also pins its clock now — its fixture was a past date, so it became time-dependent the moment behaviour started reading the wall clock.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/app/t/[teamId]/page.tsx [hunk 2]
```diff
@@ -64,6 +204,206 @@ export default async function TeamHomePage({
...
+                                        whole point of one-tap. An enum, never
+                                        a URL; see actions.ts. */}
+                                    <input type="hidden" name="from" value="home" />
+                                    <Button
+                                      type="submit"
+                                      size="sm"
+                                      variant={
+                                        state === "attending" ? "default" : "outline"
+                                      }
+                                      aria-label={`${entry.playerName} is going to ${eventHeading(event)}`}
```
[the remark this answers is not in the document]
Comment (BRCornelius) [reply src/app/t/[teamId]/page.tsx]: Confirmed and fixed in 7871cb8. Two practices in the next three events gave four buttons only two accessible names between them, so there was no way to say which control you meant.

Both labels now go through one `rsvpLabel` helper that appends `formatEventDateTime(event.startsAt)` — the same thing that distinguishes the cards visually, so the accessible name matches what a sighted user is reading. The regression test renders two practices and asserts four labels, all unique; reverting to the heading alone fails it with `expected 2 to be 4`, which is precisely the bug you described.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/lib/rsvps.ts [hunk 3]
```diff
@@ -28,6 +28,28 @@ export async function listEventRsvps(
...
+  eventIds: readonly string[],
+): Promise<EventRsvpRow[]> {
+  if (eventIds.length === 0) {
+    return [];
+  }
+
+  return db.rsvp.findMany({
+    where: { eventId: { in: [...eventIds] }, event: { teamId } },
+    select: { eventId: true, playerId: true, attending: true },
+  });
```
[the remark this answers is not in the document]
Comment (BRCornelius) [reply src/lib/rsvps.ts:50]: Agreed and added in 7871cb8 — a genuine gap, since `listEventRsvps` right above it pins exactly these things and the new function had nothing.

Four tests covering what you listed: the `eventId: { in: [...] }` filter with the `event: { teamId }` relation, the selected columns, the empty-input short-circuit (asserting the database is never called), and error propagation. The `eventId` column assertion is the load-bearing one you identified — it's what `buildRsvpStateMapsByEvent` buckets on, so dropping it from the select would collapse all three cards onto a single answer.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: AGENTS.md [hunk 4]
```diff
@@ -318,6 +320,28 @@ production — the dev command can prompt, generate new migrations, and reset th
   first week. Changing a design token or a field radius will fail `pnpm check` until the
   document is updated to match. Fix whichever side is wrong — usually the document — and
   never delete the claim to silence the test. §13 of the document explains the format.
+- **`GAME_GRACE_MS` keeps an event current for three hours after it starts, which is safe
+  for a display and not for a write.** `nextGame` and `nextEvent` both apply it, so the
+  event `/view`, `/readiness` and team home name is often one already in progress — right,
```
[the remark this answers is not in the document]
Comment (BRCornelius) [reply AGENTS.md]: Half right, and both halves are now addressed in 7871cb8.

**`nextEvent` — correct.** Stale reference to a helper I renamed to `nextEvents` in the three-event commit. Sending a contributor after a function that doesn't exist is exactly the failure this file should prevent. Fixed.

**"team home name" — not a typo.** `name` was the verb: "the event `/view`, `/readiness` and team home *name* is often one already in progress", i.e. the event those three pages name. That said, if it misparsed on reading, the sentence is doing its job badly regardless of whose fault that is — so I reworded it to "the event named by `/view`, `/readiness` and team home", which can only be read one way.

While there I also extended that note to record what your other comment surfaced: the render is not the boundary, and `rsvpAction` re-checks the start time for `from=home` posts.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: .agents/archive/issue-planner/48/task-doc.md [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+- [ ] `src/lib/calendar.ts` — add `selectNextEvent<T extends GameCandidate>(events, now)`:
+      soonest event of any type with `startsAt > now − GAME_GRACE_MS`. Refactor
+      `selectNextGame` to filter `type === "GAME"` and delegate, so "not yet finished" is
+      defined once. (Consider renaming `GameCandidate` → keep as-is; no exported-API break.)
+- [ ] `src/lib/calendar.test.ts` — cases: practice sooner than game wins; game-only lists
+      unchanged; grace window boundary; empty → null; regression: `selectNextGame` still
+      skips practices.
+- [ ] `src/lib/schedule.ts` — add `nextEvent(teamId, now = new Date())` mirroring
+      `nextGame`: same `EVENT_SELECT`, no `type` filter, no `take: 1` (same rationale
+      comment), errors NOT swallowed, returns `selectNextEvent(candidates, now)`. Extend
```
[the remark this answers is not in the document]
Comment (BRCornelius) [reply .agents/archive/issue-planner/48/task-doc.md:29]: Fixed in 7871cb8, though deliberately not the way you suggested — flagging that so you can push back.

You're right that the contradiction is confusing and shouldn't ship. But this file isn't a spec the code should match; it's the record of a plan **as approved**, before the scope was changed mid-flight to three events with per-event RSVP. `finalize-issue` archives it on merge. Rewriting the checklist to describe `nextEvents` and three cards would make it read as though that was always the plan, erasing the one genuinely useful thing it documents — that the scope moved, and when.

So instead the file keeps its checklist and gains a dated note at the top recording what changed and what the shipped contract actually is (`nextEvents(teamId, limit, now)`, `selectNextEvent` as the `limit: 1` case, buttons grouped under events, one batched read into `buildRsvpStateMapsByEvent`). Nobody reads past line 10 thinking the singular version is current, and the history stays honest.

If the convention here is that planning docs should track the code, say so and I'll rewrite it properly instead.

---
_Generated by [Claude Code](https://claude.ai/code)_

[the remark this answers is not in the document]
Comment (BRCornelius) [reply .agents/archive/issue-planner/48/task-doc.md:29]: Settled by the maintainer: planning docs under `.agents/issue-planner/` are a **record of the plan as approved**, not a spec the code should be made to match. So the checklist stays as written and keeps its scope-change note, rather than being rewritten to describe `nextEvents` and three cards.

The docs are now archived to `.agents/archive/issue-planner/48/`, where that note is the useful part of the record — it says what was planned, what shipped instead, and that the difference was a deliberate mid-flight change rather than drift.

Thanks for catching the contradiction; the disagreement was only about which half to fix.

---
_Generated by [Claude Code](https://claude.ai/code)_
